photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoSaverTask.java

```diff
@@ -0,0 +1,183 @@
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.io.File;
+import java.io.FileOutputStream;
+
+/**
+ * Created by Burhanuddin Rashid on 18/05/21.
+ *
+ * @author <https://github.com/burhanrashid52>
+ */
+class PhotoSaverTask extends AsyncTask<String, String, PhotoSaverTask.SaveResult> {
+
+    public static final String TAG = "PhotoSaverTask";
+    private @NonNull
+    SaveSettings mSaveSettings;
+    private @Nullable
+    PhotoEditor.OnSaveListener mOnSaveListener;
+    private @Nullable
+    OnSaveBitmap mOnSaveBitmap;
+    private final PhotoEditorView mPhotoEditorView;
+    private final GraphicHelper mGraphicHelper;
+    private final BrushDrawingView mBrushDrawingView;
+
+
+    public PhotoSaverTask(PhotoEditorView photoEditorView, GraphicHelper graphicHelper) {
+        mPhotoEditorView = photoEditorView;
+        mBrushDrawingView = photoEditorView.getBrushDrawingView();
+        mGraphicHelper = graphicHelper;
+        mSaveSettings = new SaveSettings.Builder().build();
+    }
+
+    public void setOnSaveListener(@Nullable PhotoEditor.OnSaveListener onSaveListener) {
+        this.mOnSaveListener = onSaveListener;
+    }
+
+    public void setOnSaveBitmap(@Nullable OnSaveBitmap onSaveBitmap) {
+        mOnSaveBitmap = onSaveBitmap;
+    }
+
+    public void setSaveSettings(@NonNull SaveSettings saveSettings) {
+        mSaveSettings = saveSettings;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+        mGraphicHelper.clearHelperBox();
+        mBrushDrawingView.destroyDrawingCache();
+    }
+
+    @SuppressLint("MissingPermission")
+    @Override
+    protected SaveResult doInBackground(String... inputs) {
+        // Create a media file name
+        if (inputs.length == 0) {
+            return saveImageAsBitmap();
+        } else {
+            return saveImageInFile(inputs[0]);
+        }
+    }
+
+    private SaveResult saveImageAsBitmap() {
+        if (mPhotoEditorView != null) {
+            return new SaveResult(null, null, buildBitmap());
+        } else {
+            return new SaveResult(null, null, null);
+        }
+    }
+
+    @NonNull
+    private SaveResult saveImageInFile(String mImagePath) {
+        File file = new File(mImagePath);
+        try {
+            FileOutputStream out = new FileOutputStream(file, false);
+            if (mPhotoEditorView != null) {
+                Bitmap capturedBitmap = buildBitmap();
+                capturedBitmap.compress(mSaveSettings.getCompressFormat(), mSaveSettings.getCompressQuality(), out);
+            }
+            out.flush();
+            out.close();
+            Log.d(TAG, "Filed Saved Successfully");
+            return new SaveResult(null, mImagePath, null);
+        } catch (Exception e) {
```

Comment:
Nit: I would narrow this `catch` block, either by catching `IOException` or creating a union of exception types using the `|` operator. Saves you from catching and squashing things like `NullPointerException`.